app/src/main/java/com/ddoczi/tasky/MainActivity.kt

```diff
@@ -49,17 +50,25 @@ fun TaskyMainScreen(
             val state by viewModel.state.collectAsStateWithLifecycle()
             LoginScreen(
                 state = state,
-                onEvent = viewModel::onEvent,
-                onSignUpClick = { navController.navigate(Route.REGISTRATION) }
+                onEvent = { event ->
+                    when(event) {
+                        is LoginEvent.OnSignUpClick -> { navController.navigate(Route.REGISTRATION) }
+                        else -> { viewModel.onEvent(event) }
+                    }
```

Comment:
There's a subtle difference between this and the snippet I shared in the last PR. You're sending the event to the ViewModel if it's NOT `OnSignUpClick` while my snippet sends it to the VM in any case.
Here, it doesn't make a difference, but generally it can often happen that you need to catch an event in both UI and ViewModel. So instead, change this to `else -> Unit` and call the `viewModel.onEvent` after the when-block